intranet/apps/eighth/views/admin/blocks.py

```diff
@@ -172,6 +106,75 @@ def add_block_view(request):
     return render(request, "eighth/admin/add_block.html", context)
 
 
+def perform_hybrid_block_signup(request, fmtdate, celery_logger):
+    blocks = EighthBlock.objects.filter(date=fmtdate)
+    block_names = {b.block_letter[4:] for b in blocks if "P1" in b.block_letter or "P2" in b.block_letter or "Virt" in b.block_letter}
+    sticky_act = EighthActivity.objects.get_or_create(name="z - Hybrid Sticky", sticky=True, administrative=True)[0]
+    failed_users = set()
+    if "P1" in block_names and "Virt" in block_names and "P2" not in block_names:
+        # sticky `virtual` and `in-person (l-z)` for `* - P1`
+        p1_blocks = {b for b in blocks if "P1" in b.block_letter}
+        for b in p1_blocks:
+            sch_act = EighthScheduledActivity.objects.create(block=b, activity=sticky_act, attendance_taken=True)
+            for g in [Group.objects.get(name="virtual"), Group.objects.get(name="in-person (l-z)")]:
+                for user in g.user_set.all():
+                    try:
+                        sch_act.add_user(user, request=request, force=True, no_after_deadline=True)
+                    except Exception:
+                        failed_users.add(user)
+
+        # sticky `in-person (a-k)` for `* - Virt`
+        virtual_blocks = {b for b in blocks if "Virt" in b.block_letter}
+        for b in virtual_blocks:
+            sch_act = EighthScheduledActivity.objects.create(block=b, activity=sticky_act, attendance_taken=True)
+            for g in [Group.objects.get(name="in-person (a-k)")]:
+                for user in g.user_set.all():
+                    try:
+                        sch_act.add_user(user, request=request, force=True, no_after_deadline=True)
+                    except Exception:
+                        failed_users.add(user)
+        celery_logger.debug("Assigned groups successfully.")
+        if failed_users:  # this shouldn't happen because only students should be in the groups, but we don't want to 500
+            celery_logger.debug(
```

Comment:
How do we want to handle errors? Because this likely won't be shown to the user and might go unnoticed in the logs

Reply:
I'm not worried about the `failed_users` error because that would only occur if a teacher or someone had been placed into one of the three groups. They shouldn't be signed up anyway, so I think that error is okay to not return. I supposed I could copy some of this logic into the initial view to output the other error messages as needed (see new force push).